src/TiledArray/permutation.h

```diff
@@ -225,6 +236,17 @@ class Permutation {
   /// \return An iterator that points to the end of the element range
   const_iterator cend() const { return p_.cend(); }
 
+  Permutation outer_permutation() const {
```

Comment:
@ryanmrichard the main reason to make this a bipartite permutation are these 2 simple methods? Can we just make them free functions and remove difficult-to-explain rationale for bipartiteness and `n_inner_`

Reply:
It's been awhile, but I feel like I tried to avoid adding this to the class and failed. There's a lot of functions which take a Permutation object. I believe there are locations where some of these functions need to operate differently for a ToT, but don't have access to the corresponding `VariableList` instance. Thus they no longer have a way to tell which modes are inner and which are outer.